libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightAlertConfig/AlertAttributeSelecxt.tsx

```diff
@@ -20,279 +20,263 @@ import {
 } from "../../../../constants/index.js";
 import { IExecutionResultEnvelope } from "../../../../../model/index.js";
 
-import { getAttributeTitle } from "./utils.js";
-import {
-    AttributeValue,
-    useAttributeValuesFromExecResults,
-} from "./hooks/useAttributeValuesFromExecResults.js";
+import { useAttributeValuesFromExecResults } from "./hooks/useAttributeValuesFromExecResults.js";
+import { getSelectedCatalogAttribute, getSelectedCatalogAttributeValue } from "./utils/getters.js";
 
 export interface IAlertAttributeSelectProps {
     execResult: IExecutionResultEnvelope | undefined;
     selectedAttribute: AlertAttribute | undefined;
     selectedValue: string | null | undefined;
     onAttributeChange: (attribute: AlertAttribute | undefined, value: string | undefined) => void;
     attributes: AlertAttribute[];
-    catalogAttributes: IAttributeMetadataObject[];
+    catalogAttributes: ICatalogAttribute[];
+    catalogDateDatasets: ICatalogDateDataset[];
 }
 
 export const AlertAttributeSelect = ({
     execResult,
-    selectedAttribute,
+    selectedAttribute: sa,
```

Comment:
sa is cryptic naming. what about `selectedAttributeProp`? Or use `effectiveSelectedAttribute` later in the code? (some convention we already use on other places)